### PR TITLE
Update to 4.14.3 and backport few more patches

### PR DESCRIPTION
**Pull request by marmarek**
QubesOS/qubes-issues#6066

### CODE REVIEW

File: patch-0001-x86-xstate-reset-cached-register-values-on-resume.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@@ -642,6 +642,13 @@ void xstate_init(struct cpuinfo_x86 *c)
+         return;
+     }
+ 
++    /*
++     * Clear the cached value to make set_xcr0() and set_msr_xss() really
++     * write it.
++     */
++    this_cpu(xcr0) = 0;
++    this_cpu(xss) = 0;
```
Comment (isodude) [inline patch-0001-x86-xstate-reset-cached-register-values-on-resume.patch:40]: I read a comment about that this should be ~0, not sure what they meant?

**Comment by jevank** [comment]
I get an issue with sys-usb while testing. Looks like pci device removing fails and it breaks vm restarting
```
2021-09-28 08:51:09.616+0000: libxl: libxl_pci.c:396:sysfs_write_bdf: Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control: No such file or directory
2021-09-28 08:51:09.780+0000: libxl: libxl_pci.c:396:sysfs_write_bdf: Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control: No such file or directory
2021-09-28 08:53:56.050+0000: libxl: libxl_pci.c:2017:do_pci_remove: Domain 23:xc_physdev_unmap_pirq irq=16: Invalid argument
2021-09-28 08:54:49.713+0000: libxl: libxl_pci.c:1493:libxl__device_pci_reset: The kernel doesn't support reset from sysfs for PCI device 0000:00:14.0
2021-09-28 08:54:49.714+0000: libxl: libxl_pci.c:396:sysfs_write_bdf: Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control: No such file or directory
2021-09-28 08:54:53.635+0000: libxl: libxl_pci.c:396:sysfs_write_bdf: Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control: No such file or directory
2021-09-28 08:54:53.800+0000: libxl: libxl_pci.c:396:sysfs_write_bdf: Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control: No such file or directory
2021-09-28 08:55:29.818+0000: libxl: libxl_pci.c:1493:libxl__device_pci_reset: The kernel doesn't support reset from sysfs for PCI device 0000:00:14.0
2021-09-28 08:56:29.392+0000: libxl: libxl_pci.c:1612:libxl__device_pci_add: Domain 6:PCI device 0000:00:14.0 already assigned to a different guest?
2021-09-28 08:56:29.392+0000: libxl: libxl_pci.c:1772:device_pci_add_done: Domain 6:libxl__device_pci_add  failed for PCI device 0:0:14.0 (rc -1)
2021-09-28 08:56:29.392+0000: libxl: libxl_create.c:1938:domcreate_attach_devices: Domain 6:unable to add pci devices
```

**Comment by jevank** [comment]
With sys-net same errors on shutdown, but no issues on restart. I use lenovo X1C6.
```
2021-09-28 09:06:32.023+0000: libxl: libxl_pci.c:2017:do_pci_remove: Domain 4:xc_physdev_unmap_pirq irq=16: Invalid argument
2021-09-28 09:06:59.608+0000: libxl: libxl_pci.c:396:sysfs_write_bdf: Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control: No such file or directory
2021-09-28 09:06:59.798+0000: libxl: libxl_pci.c:396:sysfs_write_bdf: Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control: No such file or directory
```

**Comment by marmarek** [comment]
>  Couldn't open /sys/bus/pci/drivers/pciback/allow_interrupt_control

Which kernel version in dom0? This file should be there, for some time already (5.6+?). In earlier versions, we had a custom patch with it, and the file was called `/sys/bus/pci/drivers/pciback/allow_msi_enable`. If you have it, you can ignore the above message - it tries both.

> 2021-09-28 08:56:29.392+0000: libxl: libxl_pci.c:1612:libxl__device_pci_add: Domain 6:PCI device 0000:00:14.0 already assigned to a different guest?

This looks like the key thing. Can you check `xl debug-key Q; xl dmesg` when it happens, and then see to which domain this device is assigned?

**Comment by jevank** [comment]
In my environment issue 100% reproduced on Lenovo X1C7 too. Sys-usb starts on boot and cannot be restarted.

>Which kernel version in dom0? This file should be there, for some time already (5.6+?). In earlier versions, we had a custom patch with it, and the file was called /sys/bus/pci/drivers/pciback/allow_msi_enable. If you have it, you can ignore the above message - it tries both.

5.4.148 and 0014-xen-pciback-add-attribute-to-allow-MSI-enable-flag-w.patch is included. Thanks for clarification

> This looks like the key thing. Can you check xl debug-key Q; xl dmesg when it happens, and then see to which domain this device is assigned?

(XEN) ==== PCI devices ====
(XEN) ==== segment 0000 ====
(XEN) 0000:03:00.0 - d0 - node -1  - MSIs < 126 127 128 129 130 >
(XEN) 0000:02:00.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1f.5 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1f.4 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1f.3 - d0 - node -1  - MSIs < 131 >
(XEN) 0000:00:1f.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1d.0 - d0 - node -1  - MSIs < 123 >
(XEN) 0000:00:1c.0 - d0 - node -1  - MSIs < 122 >
(XEN) 0000:00:16.0 - d0 - node -1  - MSIs < 125 >
(XEN) 0000:00:15.1 - d0 - node -1  - MSIs < >
(XEN) 0000:00:15.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:14.3 - d1 - node -1  - MSIs < 133 >
(XEN) 0000:00:14.2 - d0 - node -1  - MSIs < >
(XEN) 0000:00:14.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:12.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:08.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:04.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:02.0 - d0 - node -1  - MSIs < 124 >
(XEN) 0000:00:00.0 - d0 - node -1  - MSIs < >

Looks like device is in dom0, right?

**Comment by jevank** [comment]
Problem comes with this [commit](http://xenbits.xen.org/gitweb/?p=xen.git;a=commit;h=645fcf8a96d706b6037b7a772557213479be9a06). Reverting looks not a good idea, but Im not so deep in xen to propose fix.
What do you think?

**Comment by marmarek** [comment]
> Looks like device is in dom0, right?

Yes, and it should _not_ be. It should be assigned to a quarantine domain (`d[IO]` on the list). How does it look after successful shutdown (in your case, sys-net I guess)? Did you made any customization to a libvirt config?

**Comment by jevank** [comment]
You are right. 
>How does it look after successful shutdown (in your case, sys-net I guess)?
```
(XEN) ==== PCI devices ====
(XEN) ==== segment 0000 ====
(XEN) 0000:03:00.0 - d0 - node -1  - MSIs < 126 127 128 129 130 >
(XEN) 0000:02:00.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1f.5 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1f.4 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1f.3 - d0 - node -1  - MSIs < 131 >
(XEN) 0000:00:1f.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:1d.0 - d0 - node -1  - MSIs < 123 >
(XEN) 0000:00:1c.0 - d0 - node -1  - MSIs < 122 >
(XEN) 0000:00:16.0 - d0 - node -1  - MSIs < 125 >
(XEN) 0000:00:15.1 - d0 - node -1  - MSIs < >
(XEN) 0000:00:15.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:14.3 - d[IO] - node -1  - MSIs < >
(XEN) 0000:00:14.2 - d0 - node -1  - MSIs < >
(XEN) 0000:00:14.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:12.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:08.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:04.0 - d0 - node -1  - MSIs < >
(XEN) 0000:00:02.0 - d0 - node -1  - MSIs < 124 >
(XEN) 0000:00:00.0 - d0 - node -1  - MSIs < >
```

I tested with commit reverted and problem fixed.

**Comment by jevank** [comment]
I mean with commit reverting USB controller shown as d[IO] domain and restarting is ok.

**Comment by jevank** [comment]
> Did you made any customization to a libvirt config?
not

**Comment by marmarek** [comment]
Can you reboot with `iommu=debug` on Xen cmdline (and maybe `loglvl=all` too), and post all the Xen messages + during start and shutdown of both VMs (+`xl debug-key Q` each time)?

**Comment by jevank** [comment]
Sure, but only tomorrow

**Comment by jevank** [comment]
here it is:
- boot
- sys-usb start
- sys-usb shutdow
- sys-net start
- sys-net shutdown
- sys-usb start

https://gist.github.com/jevank/d03f94a4cadc3cdffc4069c651157b88

**Comment by marmarek** [comment]
PipelineRetry

**Pull request merged**